node/src/proxy_server/mod.rs

```diff
@@ -1020,6 +1197,35 @@ impl StreamKeyFactory for StreamKeyFactoryReal {
     }
 }
 
+#[derive(Clone, Debug, Eq, PartialEq)]
+struct DNSFailureRetry {
+    unsuccessful_request: ClientRequestPayload_0v1,
+    retries_left: usize,
+}
+
+#[derive(Clone, Debug, Eq, PartialEq)]
+struct Hostname {
+    hostname: String,
+}
+
+impl Hostname {
+    #[allow(dead_code)]
+    fn new(raw_url: &str) -> Self {
```

Comment:
Is `new` the best name for this constructor function? How about implementing `FromStr` and using `from_str` to make a `Hostname` out of a URL? Or maybe not implementing `FromStr` at all, but having the `impl Hostname` section have something called `from_url(raw_url: &str) -> Self` that does what `new()` does now?